core/src/main/scala/com/stripe/agate/tensor/OnnxNumber.scala

```diff
@@ -525,7 +525,10 @@ object OnnxNumber {
     }
 
   def forDataType(dt: DataType): OnnxNumber[dt.Elem] =
-    forDataTypeMap(dt)
+    forDataTypeMap.get(dt: DataType.Aux[dt.Elem]) match {
+      case Some(num) => num
+      case None      => throw new IllegalArgumentException(s"not a numeric datatype=$dt")
```

Comment:
Just chucking an exception requires the least amount of change. We could update our numeric operations on tensor to return `Try`s instead, but figured I'd see how y'all feel about it.